rust-symcrypt/src/ecdh.rs

```diff
@@ -0,0 +1,270 @@
+/* EcDh functions. For further documentation please refer to symcrypt.h */
+
+use crate::curve_type::*;
+use crate::errors::SymCryptError;
+use std::vec;
+use symcrypt_sys;
+
+/// EcDhKey is a wrapper around symcrypt_sys::PSYMCRYPT_ECKEY, this is to let rust handle the free'ing of this pointer
+/// EcDhKey must be allocated after EcDhExpandedCurve and free'd before EcDhExpandedCurve is free'd
+struct EcDhKey {
+    inner: symcrypt_sys::PSYMCRYPT_ECKEY,
+    expanded_curve: EcDhExpandedCurve,
+}
+
+struct EcDhExpandedCurve(symcrypt_sys::PSYMCRYPT_ECURVE);
```

Comment:
```suggestion
struct EcDhExpandedCurve(NonNull<symcrypt_sys::SYMCRYPT_ECURVE>);
```

Same thing here, you only ever construct this with non null pointers, so you can tell the type system that's the case.